app.js

```diff
@@ -137,38 +138,167 @@ class Products {
+      }
+      //for the increament of the variant
+      else if (event.target.classList.contains("fa-chevron-up")) {
+        let increaseNumber = event.target;
+        let id = increaseNumber.dataset.id;
+        let tempItem = cart.find((item) => item.id === id);
+        tempItem.amount = tempItem.amount + 1;
+        Storage.saveCart(cart);
+        this.setCartValues(cart);
+        increaseNumber.nextElementSibling.innerText = tempItem.amount;
+      }
+      //for the reduction of the variant
+      else if (event.target.classList.contains("fa-chevron-down")) {
+        let lowerAmount = event.target;
+        let id = lowerAmount.dataset.id;
+        let tempItem = cart.find((item) => item.id === id);
+        tempItem.amount = tempItem.amount - 1;
+        if (tempItem.amount > 0) {
+          Storage.saveCart(cart);
+          this.setCartValues(cart);
+          lowerAmount.previousElementSibling.innerText = tempItem.amount;
+        } else {
+          cartContent.removeChild(lowerAmount.parentElement.parentElement);
+          this.removeItem(id);
+        }
+      }
+    });
+    //For the clear cart  button event
+    //clearCartBtn.addEventListener('click',() => {
+    //this.clearCart();
+    //});
+  }
+  //For the Clear cart button Functionalities.
+  //clearCart() {
+  //let cartItems = cart.map(item => item.id);
+  //cartItems.forEach(id => this.removeItem(id))
+
+  //while(cartContent.children.length >0) {
+  //  cartContent.removeChild(cartContent.children[0])
+  // }
+  // this.hideCart();
+  //}
+
+  //to remove the items from the cart doing that with their ID.
+  removeItem(id) {
+    cart = cart.filter((item) => item.id !== id);
+    this.setCartValues(cart);
+    Storage.saveCart(cart);
+    let button = this.getSingleButton(id);
+    button.disabled = false;
+    button.innerHTML = `<i class ="fas fa-shopping-cart">add to cart</i>`;
+  }
+  getSingleButton(id) {
+    return buttonsDOM.find((button) => button.dataset.id === id);
+  }
+}
+
+//To store it in the local storage.
+
+class Storage {
+  static saveProducts(products) {
+    localStorage.setItem("products", JSON.stringify(products));
+  }
+  static getProduct(id) {
+    let products = JSON.parse(localStorage.getItem("products"));
+    return products.find((product) => product.id === id);
+  }
+  //Getting the new cart values and saving it in the local storage
+  static saveCart(cart) {
+    localStorage.setItem("cart", JSON.stringify(cart));
+  }
+  static getCart() {
+    return localStorage.getItem("cart")
+      ? JSON.parse(localStorage.getItem("cart"))
+      : [];
   }
+}
+
+document.addEventListener("DOMContentLoaded", () => {
+  const ui = new UI();
+  const products = new Products();
+
+  // To setup the products to load in cart
+  ui.setupAPP();
+
+  //get all products to the local storage.
+  products
+    .getProducts()
+    .then((products) => {
+      ui.displayProducts(products);
+      Storage.saveProducts(products);
```

Comment:
are you trying to cache the data here ?
if yes... (tell me why, would like to know)
data caching should be at the request level ie `getProducts`
also, you need to consider what happens when the data change. 
to make an informed decision, you need to think about the following 
-  how do I know the data changed?
-  how do I invalidate and cache the data when the data integrity is intact? 

if No, tell me what you are trying to do?

Reply:
I'm trying to get all the products to the local storage by running that operation.the then method chains the display products and also get it to the local storage